diff --git a/src/scripts/xzgrep.in b/src/scripts/xzgrep.in
index 4a1fc1e..9bc5656 100644
--- a/src/scripts/xzgrep.in
+++ b/src/scripts/xzgrep.in
@@ -27,7 +27,7 @@
 xz='@xz@ --format=auto'
 unset GZIP BZIP BZIP2
 
-case ${0##/*} in
+case ${0##*/} in
   *egrep*) prog=xzegrep; grep=${GREP:-egrep};;
   *fgrep*) prog=xzfgrep; grep=${GREP:-fgrep};;
   *)       prog=xzgrep; grep=${GREP:-grep};;
@@ -35,7 +35,7 @@ esac
 
 version="$prog (@PACKAGE_NAME@) @VERSION@"
 
-usage="Usage: ${0##/*} [OPTION]... [-e] PATTERN [FILE]...
+usage="Usage: ${0##*/} [OPTION]... [-e] PATTERN [FILE]...
 Look for instances of PATTERN in the input FILEs, using their
 uncompressed contents if they are compressed.
 
